File: helidon-server/src/main/java/com/programacion/web/db/Post.java
package com.programacion.web.db;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;
import lombok.extern.jackson.Jacksonized;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Post {

    private Integer id;
    private String title;
    private String body;
    private User user;
    private List<Comment> comments;

}

File: FrontendReact/src/models/Post.ts
import type { User } from "./User";

export interface Post {
    id: number;
    title: string;
    body: string;
    user?: User;
    comments?: Comment[];
}

export interface Comment {
    id: number;
    name: string;
    email: string;
    body: string;
    post?: Post;
}
